### ISSUE TITLE
Release of a new version with updated dependencies

### ISSUE THREAD

**Issue by elenaf9** [mcginty/snow#119]
The latest version of this library on crates.io (v0.7.2) has outdated dependencies for `aes-gcm` (uses v.0.7 instead of the latest version v0.9) and `chacha20poly1305` (uses v0.6.0 instead of v0.8.0) which results dependencies to the unmaintained crates [`block-cipher`](https://github.com/iotaledger/stronghold.rs/issues/72), [`stream-cipher`](https://github.com/iotaledger/stronghold.rs/issues/74), [`aes-soft`](https://github.com/iotaledger/stronghold.rs/issues/207) and [`aesni`](https://github.com/iotaledger/stronghold.rs/issues/208).
I saw that there are dependabot PRs (#117 and #118) that would update these dependencies and that you guys also already published a version 0.7.3 in February, but for some reason that release was yanked and there has not been a new one since then.

**Comment by mcginty** [comment]
Yeah! So sorry about that - it got yanked because the update broke API compatibility and I was waiting for the updated `rand` release from https://github.com/dalek-cryptography/x25519-dalek, which seems to still be pending. Given that, I'll absolutely just release a new minor version now lest I get distracted again and everybody get stuck for even more time...
